### ISSUE TITLE
Conditionals: string is allowed as conditional statement

### ISSUE THREAD

**Issue by wiibaa** [elastic/logstash#1589]
Current grammar allows using string as conditionals statements like

```
if "i'm a fool" {
  noop {
    add_tag => "fool"
  }
}
```

But as reported in LOGSTASH-1434, this could be confusing to non-programmer/distracted users that would write 

```
if [field] == "string1" or "string2" {
  ....
}
```

and do not understand why the condition is always matched...

**Comment by jordansissel** [comment]
Still something we could work on, but a low priority for me.  I am open to patches that improve this. :)

**Comment by guyboertje** [comment]
I don't think we should make syntax changes before the LIR changes #6018 are introduced.

@andrewvc said that for debugging a truthy constant value could be used to force an conditionl block to execute.

Is this feature (for LIR) we want to turn off by default and should be enabled for debugging to prevent accidental misuse?
